common/djangoapps/edxmako/request_context.py

```diff
@@ -45,31 +38,36 @@ def get_template_context_processors():
     return tuple(import_string(path) for path in context_processors)
 
 
-def get_template_request_context():
+def get_template_request_context(request=None):
     """
     Returns the template processing context to use for the current request,
     or returns None if there is not a current request.
     """
-    request = getattr(REQUEST_CONTEXT, "request", None)
-    if not request:
-        return None
 
-    request_cache_dict = RequestCache.get_request_cache().data
-    cache_key = "edxmako_request_context"
+    request_cache_dict = request_cache.get_cache('edxmako')
+    cache_key = "request_context"
     if cache_key in request_cache_dict:
         return request_cache_dict[cache_key]
 
+    if request is None:
+        request = get_current_request()
+
+    if request is None:
+        return None
+
     context = RequestContext(request)
-    context['is_secure'] = request.is_secure()
-    context['site'] = safe_get_host(request)
 
-    # This used to happen when a RequestContext object was initialized but was
-    # moved to a different part of the logic when template engines were introduced.
-    # Since we are not using template engines we do this here.
-    # https://github.com/django/django/commit/37505b6397058bcc3460f23d48a7de9641cd6ef0
-    for processor in get_template_context_processors():
-        context.update(processor(request))
+    if request is not None:
```

Comment:
This check and indentation isn't necessary, is it? (Since the block above checks if `request` is `None` and returns if so.)